lib/bounces.js

```diff
@@ -39,22 +57,34 @@ module.exports = function (log) {
       else if (message.complaint && message.complaint.complaintFeedbackType === 'abuse') {
         recipients = message.complaint.complainedRecipients
       }
-      for (var i = 0; i < recipients.length; i++) {
-        var email = recipients[i].emailAddress
-        log.info({ op: 'handleBounce', email: email, bounce: !!message.bounce })
-        log.increment('account.email_bounced')
-        db.emailRecord(email)
-          .done(
-            deleteAccountIfUnverified,
-            gotError.bind(null, email)
-          )
-      }
-      message.del()
+      var p = P.resolve()
+      recipients.forEach(function(recipient) {
+        var email = recipient.emailAddress
+        p = p.then(
```

Comment:
This is an old habit I have for unrolling a for-loop into a promise chain, I'll check out the Promise helper functions to find something cleaner.  The idea is for this function to return a promise that only resolves when all the bounces have been dealt with, to make things deterministic for the tests.